src/V3AstNodeDType.h

```diff
@@ -746,6 +746,9 @@ class AstEnumDType final : public AstNodeDType {
         widthFromSub(subDTypep());
     }
     ASTGEN_MEMBERS_AstEnumDType;
+
+    using TableMap = std::map<VAttrType, AstVar*>;
+    TableMap m_tableMap;  // Created tables so can remove duplicates
     const char* broken() const override {
```

Comment:
The broken method should iterate over the entire map and return an error if brokeExists() is false on any of the stored pointers.  This indicates a dangling pointer.

Reply:
No, the function must return false if any brokeExists are false.  Note the ! in the beginning, so your test will mis-succeed if any pointer is good, it needs to be "|| all_are_true_of (brokeExists())"

Reply:
Actually I'd suggest add a new line.

        BROKEN_RTN(std::any_of(m_tableMap.begin(), m_tableMap.end(),
                                    [](const auto& p) { return !p.second->brokeExists(); })));